docs/project-file-structure.md

```diff
@@ -64,30 +91,139 @@ to organize their structure
 
+A scene and its script should always be together within the directory i.e. no /scripts/ and /scenes/ directories should exist within
+the project somewhere.
+
 Example:
 ```
 root/
 |- src/
-    |- Theme/
+    |- theme/
     |   |- ButtonNormal.tres
     |   |- ButtonHover.tres
     |   |- ButtonDisabled.tres
-    |- Scripts/
+    |- utility/
     |   |- Math.gd
-    |- Menu/
+    |- menu/
     |   |- MenuManager.tscn
     |   |- MenuManager.gd
     |   |- MainMenu/
     |   |   |- MainMenu.tscn
     |   |   |- MainMenu.gd
     |   |   |- MainMenuFont.ttf
     |   |   |- MainMenuLabel.tres
-    |   |- SettingsMenu/
+    |   |- settings_menu/
     |   |   |- Settings.tscn
     |   |   |- Settings.gd
     |   |   |- Resolution/
     |   |   |   |- ResolutionSelector.gd
-    |   |   |- Volume/
+    |   |   |- volume/
     |   |   |   |- VolumeGrid.tscn
     |   |   |   |- VolumeGrid.gd
 ```
+
+#### Naming Convention
+
+Directory names should be snake_case.
+```
+root/src/game_session/
+```
+
+File names should be PascalCase.
+```
+root/src/game_session/Map.tscn
+```
+
+### Asset Organization
+
+Note: Asset files should not be exported with Godot in a pck file. They should be downloaded via an installer or handled in a similar
+fashion to the data files and their exclusion from the export
+
+Asset files should be organized within a relevant category from root/
+```
+root/art/
+root/music/
+root/sfx/
+```
+
+Asset files should be organized in a subdirectory that specifies it's relevancy within the application.
+```
+root/art/economy/
+root/art/military/
+root/sfx/economy/
+root/sfx/technology/
+```
+
+Asset files should be prefixed by the asset type
+```
+iconCoal.png
+modelSoldier.obj
+ostTheCrown.mp3
+videoIntro.mp4
+```
+
+The prefix can be removed if the file is within a subdirectory that replaces the asset type
+```
+./iconCoal.png -> ./icon/Coal.png
+./ostTheCrown.png -> ./ost/TheCrown.mp3
+```
+
+#### Naming Convention
+
+Directories should be snake_case.
+
+Asset files should be PascalCase for their name.
+```
+TheCrown.mp3
+Coal.png
+```
+
+Asset files with a prefix should have that prefix be lowercase.
+```
+ostTheCrown.mp3
+iconCoal.png
+```
+
+### Data File Organization
+
+All data files should be in the root/common/ directory.
+
+Data files should be organized by category.
+```
+root/common/map/
+root/common/technology/
+root/common/pop/
+```
+
+Data files that are plain text should have the ov2d file extension.
+```
+root/common/pop/soldier.ov2d
+```
+
+Data files that are complex data like a bitmap should make use of the standard file extension for that data
+```
+root/common/map/provinces.bmp
+```
+
+Data files can be further organized by subdirectory or prefix.
+```
+root/common/economy/goods/Coal.ov2d
+root/common/economy/goodsCoal.ov2d
+```
+
+#### Naming Convention
+
+Directories should be snake_case.
+```
+root/common/economy_technology/
+```
```

Comment:
````suggestion
Data file subdirectories should be lower case.
```
root/common/economy/technology/
```
````

Reply:
Is there a reason to distinguish between lower case and snake case? There might still be valid cases that have 2 words (that I am conveniently blanking on).

Reply:
Changing lower case to snake case for the minor possibility that there is a valid case of a directory needing two words i.e. its not worth the distinction of lower case to snake case